### PR TITLE
CFD-290 : Add "V&V" to homepage's activity stream.

### PR DESCRIPTION
**Pull request by nader77**
#484

### CODE REVIEW

File: capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 1]
```diff
@@ -881,13 +878,38 @@ function c4m_message_load_homepage_activity_stream() {
...
+
+  $tables['field_data_field_group_node1']['join type'] = 'LEFT';
+  $values = $conditions[1]['value'];
+  $or = db_or()
+    ->condition('field_group_node_target_id',  $values, 'IN')
+    ->isNull('field_group_node_target_id');
+   $query->condition($or);
+
+  foreach ($conditions as $key => $condition) {
+    if ($conditions['field'] == 'field_data_field_group_node1.field_group_node_target_id') {
```
Comment (HelenaEksler) [inline capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: Looks wrong.
``$conditions['field']`` => ``$condition['field']``

File: capacity4more/modules/c4m/message/c4m_message/c4m_message.module [hunk 2]
```diff
@@ -881,13 +878,38 @@ function c4m_message_load_homepage_activity_stream() {
...
+      unset($conditions[$key]);
```
Comment (HelenaEksler) [inline capacity4more/modules/c4m/message/c4m_message/c4m_message.module]: This looks like we are removing the condition that we've just added in the db_or.

File: capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/node/activity_stream/1.0/C4mRestfulActivityStreamResource.class.php [hunk 3]
```diff
@@ -56,9 +47,19 @@ public function viewEntity($entity_id) {
    * Display only published entities in the activity stream.
    */
   public function getQueryForList() {
+    $request = $this->getRequest();
     $query = parent::getQueryForList();
 
     $query->fieldCondition('field_entity_published', 'value', 1);
+
+    if (!empty($request['group']) && intval($request['group'])) {
```
Comment (HelenaEksler) [inline capacity4more/modules/c4m/restful/c4m_restful/plugins/restful/node/activity_stream/1.0/C4mRestfulActivityStreamResource.class.php]: Add comments.

**Comment by HelenaEksler** [comment]
Got errors
selection_206

Can't see in the homepage any V&V messages.

**Comment by nader77** [comment]
The errors are because the line comment you made, ``conditions``.

Please update some article or add some article and check if you can see V&V messages.

**Comment by HelenaEksler** [comment]
Updated  one article and created one. Still can't see V&V messages.

**Comment by nader77** [comment]
Ok checking.

**Comment by nader77** [comment]
@HelenaEksler Fixed all your comments and in result the V&V appeared in the activity stream.

**Comment by HelenaEksler** [comment]
@nader77 Reviewed. Looks good.

**Comment by zero2one** [comment]
@nader77 One test is broken:
See: https://travis-ci.org/capacity4dev/capacity4more/jobs/50644313#L1404

**Comment by nader77** [comment]
@zero2one Test is fixed.

**Pull request merged**